types/dockerode/index.d.ts

```diff
@@ -832,18 +833,18 @@ declare class Dockerode {
   createContainer(options: Dockerode.ContainerCreateOptions, callback: Callback<Dockerode.Container>): void;
   createContainer(options: Dockerode.ContainerCreateOptions): Promise<Dockerode.Container>;
 
-  createImage(options: {}, callback: Callback<Dockerode.Image>): void;
-  createImage(auth: any, options: {}, callback: Callback<Dockerode.Image>): void;
-  createImage(options: {}): Promise<Dockerode.Image>;
-  createImage(auth: any, options: {}): Promise<Dockerode.Image>;
+  createImage(options: {}, callback: Callback<any>): void;
```

Comment:
Why are you changing these `any`?

Reply:
After digging into this, it looks like it returns a `NodeJS.ReadableStream`

Reply:
As I outlined in my PR message, it actually returns an 'HTTP.IncomingMessage', but I changed it to 'any' to be consistent with all the other methods which also just return 'any'. I prefer consistency when changing code and believe all of the 'any' return types should be aligned for now and then fixed together later. This is out of scope for this PR.

Reply:
The API surface is pretty large in `dockerode` so I prefer to just chip away at them as I use that part of the API. I originally submitted these types, I simply didn't have the time or patience to correctly define everything.
In this case, I just got it wrong and we know what the correct type is. Luckily here it's just a `node` type and not a `Docker Remote API` type, so it's pretty easy to change it.

Reply:
I've added these return types for this area only. I am assuming no other type is ever returned from these methods as I've only tested it using promises and reading as a stream.